### PR TITLE
Top m

### PR DESCRIPTION
**Pull request by Hot00Tea**

### CODE REVIEW

**Review by MentorMateAcademy — CHANGES_REQUESTED** [review]
Your task not finished

**Comment by Hot00Tea** [comment]
Screenshot_1
 
В мене помилка яку я не знаю як виправити

**Comment by Hot00Tea** [comment]
Я здав чи треба переробити щось?

**Comment by Hot00Tea** [comment]
Done

File: src/main/java/core/basesyntax/Truck.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    @Override
+    public void doWork() {
+
+    }
+
+    @Override
+    public void stopWork() {
+
+    }
+
```
Comment (okuzan) [inline src/main/java/core/basesyntax/Truck.java:14]: add implementation

File: src/main/java/core/basesyntax/Truck.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package core.basesyntax;
+
+public class Truck extends Machine {
+
```
Comment (okuzan) [inline src/main/java/core/basesyntax/Truck.java:4]: don't leave an emtpy line after method/class definition

File: src/main/java/core/basesyntax/MainApp.java [hunk 3]
```diff
@@ -2,4 +2,16 @@
 
 public class MainApp {
 
+    private Excavator excavator = new Excavator();
+    private Truck truck = new Truck();
+    private Bulldozer bulldozer = new Bulldozer();
+    private Machine[] machines = {excavator, truck, bulldozer};
```
Comment (okuzan) [inline src/main/java/core/basesyntax/MainApp.java:8]:
```suggestion
    private Machine[] machines = {new Excavator(), new Truck(), new Bulldozer()};
```

**Review by okuzan — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/Truck.java (hunk 1 above)
Comment (Hot00Tea) [reply src/main/java/core/basesyntax/Truck.java:14]: Не розумію як

File: src/main/java/core/basesyntax/Bulldozer.java [hunk 4]
```diff
@@ -0,0 +1,14 @@
+package core.basesyntax;
+
+public class Bulldozer extends Machine {
+
+    @Override
+    public void doWork() {
+
```
Comment (liliia-ponomarenko) [inline src/main/java/core/basesyntax/Bulldozer.java:7]: Let's add the realization to all override methods in classes that extend Machine 
for example: 
```suggestion
 System.out.println("Bulldozer started working!");
```

File: src/main/java/core/basesyntax/Bulldozer.java [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+public class Bulldozer extends Machine {
+
+    @Override
+    public void doWork() {
+
+    }
+
+    @Override
+    public void stopWork() {
+
```
Comment (liliia-ponomarenko) [inline src/main/java/core/basesyntax/Bulldozer.java:12]: add realization

File: src/main/java/core/basesyntax/Excavator.java [hunk 6]
```diff
@@ -0,0 +1,13 @@
+package core.basesyntax;
+
+public class Excavator extends Machine {
+
+    @Override
+
```
Comment (liliia-ponomarenko) [inline src/main/java/core/basesyntax/Excavator.java:6]: redundant empty line 
```suggestion
```

File: src/main/java/core/basesyntax/Machine.java [hunk 7]
```diff
@@ -0,0 +1,9 @@
+package core.basesyntax;
+
+public abstract class Machine {
+
+    abstract void doWork();
+
+    abstract void stopWork();
+}
+
```
Comment (liliia-ponomarenko) [inline src/main/java/core/basesyntax/Machine.java:9]: redundant empty line 
```suggestion
```

File: src/main/java/core/basesyntax/MainApp.java [hunk 8]
```diff
@@ -2,4 +2,13 @@
...
+    private Machine[] machines = {new Excavator(), new Truck(), new Bulldozer()};
+
+    public void main() {
+        for (int i = 0; i < machines.length; i++) {
+            machines[i].doWork();
+            machines[i].stopWork();
+        }
+    }
 }
+
```
Comment (liliia-ponomarenko) [inline src/main/java/core/basesyntax/MainApp.java:14]: redundant empty line

File: src/main/java/core/basesyntax/Excavator.java [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+    @Override
+
+    public void doWork() {
+    }
+
+    @Override
+    public void stopWork() {
+    }
+}
```
Comment (liliia-ponomarenko) [inline src/main/java/core/basesyntax/Excavator.java:13]: add realization

File: src/main/java/core/basesyntax/Truck.java [hunk 10]
```diff
@@ -0,0 +1,13 @@
...
+public class Truck extends Machine {
+
+    @Override
+    public void doWork() {
+    }
+
+    @Override
+    public void stopWork() {
+    }
```
Comment (liliia-ponomarenko) [inline src/main/java/core/basesyntax/Truck.java:11]: add realization

File: src/main/java/core/basesyntax/Truck.java [hunk 11]
```diff
@@ -0,0 +1,13 @@
...
+
```
Comment (liliia-ponomarenko) [inline src/main/java/core/basesyntax/Truck.java:12]: redundant empty line

**Review by liliia-ponomarenko — CHANGES_REQUESTED** [review]
Let's add realization to your solution ;)

File: src/main/java/core/basesyntax/Bulldozer.java [hunk 12]
```diff
@@ -0,0 +1,14 @@
+package core.basesyntax;
+
+public class Bulldozer extends Machine {
+
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/Bulldozer.java:4]: [Please don't add redundant empty lines to your code.](https://github.com/mate-academy/jv-oop/blob/master/checklist.md#please-dont-add-redundant-empty-lines-to-your-code)

File: src/main/java/core/basesyntax/Excavator.java [hunk 13]
```diff
@@ -0,0 +1,14 @@
+package core.basesyntax;
+
+public class Excavator extends Machine {
+
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/Excavator.java:4]: same comment

File: src/main/java/core/basesyntax/Machine.java [hunk 14]
```diff
@@ -0,0 +1,8 @@
+package core.basesyntax;
+
+public abstract class Machine {
+
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/Machine.java:4]: same comment

File: src/main/java/core/basesyntax/MainApp.java [hunk 15]
```diff
@@ -2,4 +2,12 @@
 
 public class MainApp {
 
+    private Machine[] machines = {new Excavator(), new Truck(), new Bulldozer()};
+
+    public void main() {
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/MainApp.java:7]:
```suggestion
    public static void main(String[] args) {
```

File: src/main/java/core/basesyntax/MainApp.java [hunk 16]
```diff
@@ -2,4 +2,12 @@
 
 public class MainApp {
 
+    private Machine[] machines = {new Excavator(), new Truck(), new Bulldozer()};
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/MainApp.java:5]: replace to main method

File: src/main/java/core/basesyntax/Truck.java [hunk 17]
```diff
@@ -0,0 +1,14 @@
+package core.basesyntax;
+
+public class Truck extends Machine {
+
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/Truck.java:4]: [Please don't add redundant empty lines to your code.](https://github.com/mate-academy/jv-oop/blob/master/checklist.md#please-dont-add-redundant-empty-lines-to-your-code)

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/Bulldozer.java (hunk 12 above)
Comment (Elena-Bruyako) [reply src/main/java/core/basesyntax/Bulldozer.java:4]: Do you have some questions about this comment? [Please don't add redundant empty lines to your code.](https://github.com/mate-academy/jv-oop/blob/master/checklist.md#please-dont-add-redundant-empty-lines-to-your-code)

File: src/main/java/core/basesyntax/Bulldozer.java [hunk 18]
```diff
@@ -0,0 +1,15 @@
...
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/Bulldozer.java:4]: remove 4 line

File: src/main/java/core/basesyntax/Bulldozer.java [hunk 19]
```diff
@@ -0,0 +1,15 @@
+package core.basesyntax;
+
+public class Bulldozer extends Machine {
+
+    @Override
+    public void doWork() {
+        System.out.println("Bulldozer started working!");
+
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/Bulldozer.java:8]: remove 8 line

File: src/main/java/core/basesyntax/Excavator.java (hunk 13 above)
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/Excavator.java:4]: remove 4 line

File: src/main/java/core/basesyntax/Truck.java (hunk 17 above)
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/Truck.java:4]: remove 4 line

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]

**Review by Elena-Bruyako — APPROVED** [review]

**Pull request closed without merging**